pytorch_lightning/utilities/optimizer.py

```diff
@@ -29,5 +29,7 @@ def optimizers_to_device(optimizers: Iterable[Optimizer], device: _DEVICE) -> No
 
 def optimizer_to_device(optimizer: Optimizer, device: _DEVICE) -> None:
     """Moves the state of a single optimizer to the device."""
-    for p, v in optimizer.state.items():
-        optimizer.state[p] = apply_to_collection(v, torch.Tensor, move_data_to_device, device)
+    state, param_groups = optimizer.state_dict().values()
+    for p, v in state.items():
+        state[p] = apply_to_collection(v, torch.Tensor, move_data_to_device, device)
+    optimizer.load_state_dict({"state": state, "param_groups": param_groups})
```

Comment:
This might work instead:
```suggestion
    state_dict = optimizer.state_dict()
    state_dict = apply_to_collection(state_dict, torch.Tensor, move_data_to_device, device)
    optimizer.load_state_dict(state_dict)
```
```